gtsam/navigation/CombinedImuFactor.h

```diff
@@ -86,24 +100,25 @@ PreintegrationCombinedParams(const Vector3& n_gravity) :
   const Matrix6& getBiasAccOmegaInt() const { return biasAccOmegaInt; }
   
 private:
-  /// Default constructor makes unitialized params struct
-  PreintegrationCombinedParams() {}
 
   /** Serialization function */
   friend class boost::serialization::access;
   template <class ARCHIVE>
-    void serialize(ARCHIVE& ar, const unsigned int /*version*/) {
-    ar& BOOST_SERIALIZATION_BASE_OBJECT_NVP(PreintegratedRotation::Params);
-    ar& BOOST_SERIALIZATION_NVP(biasAccCovariance);
-    ar& BOOST_SERIALIZATION_NVP(biasOmegaCovariance);
-    ar& BOOST_SERIALIZATION_NVP(biasAccOmegaInt);
+  void serialize(ARCHIVE& ar, const unsigned int /*version*/) {
+    namespace bs = ::boost::serialization;
+    ar & BOOST_SERIALIZATION_BASE_OBJECT_NVP(PreintegrationParams);
+    ar & bs::make_nvp("biasAccCovariance",
+         bs::make_array(biasAccCovariance.data(), biasAccCovariance.size()));
```

Comment:
Is this because Eigen is not correctly serialized?

Reply:
Yes. From my understanding and what I have seen, we need to extract the underlying data array for Boost.

Reply:
Yeah, but why is the Matrix serialization in Matrix.h not triggered? Perhaps because these are fixed-size matrices? If so, maybe the solution for this *and all other future issues* is to change the signature of those functions to

```
    template<class Archive, typename Derived>
    void save(Archive & ar, const Eigen::MatrixBase<Derived>& m, unsigned int /*version*/) {
```

Does that work?

Reply:
Oh I didn't see that the serialization is defined in `Matrix.h`. From the looks of it, yes, the serialization definition is for dynamic matrices, and these (especially `preintMeasCov_`) are static, so Eigen is probably getting confused.

That should be a separate PR IMO, but I will test it out here first.

Reply:
So I had to go with `Eigen::PlainObjectBase` since `MatrixBase` doesn't support the `data()` method (strange?). Waiting on a local test(s) to pass before I create the PR.

Reply:
Actually that didn't work. I ended up doing something different which worked very nicely. Creating the PR now.